### ISSUE TITLE
Metadata has no NavigationPropertyBinding for contained navigation property.

### ISSUE THREAD

**Issue by lewischeng-ms** [OData/odata.net#312]
Spec says:

```
If the value of the Target attribute is a TargetPath, it MUST resolve to an entity set, singleton, or containment navigation property in scope.
```

But now CsdlWriter will only write the target name instead of a path:
https://github.com/OData/odata.net/blob/master/src/Microsoft.OData.Edm/Csdl/Serialization/EdmModelCsdlSchemaWriter.cs#L722

In addition, navigation property binding for contained navigation properties are not created somehow. Please investigate.

Thanks @abkmr for reporting the issue.

**Comment by sab39** [comment]
Something like [this](https://github.com/sab39/odata.net/compare/ODataV4-6.x...sab39:sab39-contained-entitysets)? (note: I threw that together quickly by editing directly on github, so it's not tested even to the point of making sure it compiles - it's just to show the general concept).

If I understand right, you can already add the bindings to the model, by code like:

``` C#
var containedEntitySet = (EdmNavigationSource)rootEntitySet.FindNavigationTarget(containedNavigationProperty);
containedEntitySet.AddNavigationTarget(subNavigationProperty, targetEntitySet);
```

If I'm understanding the code right, my changes to EdmModelCsdlSchemaWriter and EdmModelCsdlSerializationVisitor would be sufficient for those bindings to get emitted to the CSDL correctly.

**Comment by luoyan0517** [comment]
@sab39 We really appreciated for your contribution. 
For your implementation, the only concern is that every time we want to get the navigation target of the `subNavigationProperty`, we at least call FindNavigationTarget 2 times. And for deep containment scenario, it will be more complex. Another thing is that although we call AddNavigationTarget on contained entityset, the path would be written on the root entityset, right? It seems kind of inconsistent. 

Actually we are able to add navigation property for containment. It is added and found through root entity set. Like `personSet.AddNavigationTarget(flightAirlineNavigation, airlineSet);` in [Trippin service](https://github.com/OData/odata.net/blob/ODataV4-6.x/test/EndToEndTests/Services/ODataTripPinService/TripPinInMemoryModel.cs). The only thing is that the output binding path format in [CSDL](http://services.odata.org/V4/(S(qqntzoewadope25a3bh2d5bi))/TripPinServiceRW/$metadata) is not right, but it does not hurt the usage.

Besides, FYI, we have fixed this in ODataLib 7.0, tracked by #629,  and documentation is [here](http://odata.github.io/odata.net/v7/#06-21-Multi-Binding). So different from your implementation, the navigation target is still added from entityset which containment belongs to, with a specific path (TargetPath). Take the [Trippin Service](http://services.odata.org/V4/(S(qqntzoewadope25a3bh2d5bi))/TripPinServiceRW/$metadata) for example, to add the binding for Airline, the code is like:
`var path = new EdmPathExpression("Trips/PlanItems/Microsoft.OData.SampleService.Models.TripPin.Flight/Airline");
personSet.AddNavigationTarget(flightAirlineNavigation, airlineSet, path);`

And FindNavigationTarget is done on personSet as well, with the same path.

**Comment by habbes** [comment]
Closing this issue due to inactivity. If this issue is still a concern in the latest stable version of the library, kindly re-open or create a new issue.
